src/block_cache/block_chunkcache.c

```diff
@@ -628,15 +646,73 @@ __wt_chunkcache_remove(
     }
 }
 
+/*
+ * __wt_chunkcache_reconfig --
+ *     Re-configure the chunk cache.
+ */
+int
+__wt_chunkcache_reconfig(WT_SESSION_IMPL *session, const char **cfg)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
+    WT_CONFIG_ITEM cval;
+    WT_DECL_RET;
+    char **old_pinned_list, **pinned_objects;
+    unsigned int cnt, i;
+
+    chunkcache = &S2C(session)->chunkcache;
+    old_pinned_list = chunkcache->pinned_objects.array;
+    pinned_objects = NULL;
+    cnt = 0;
+
+    /* When reconfiguring, check if there are any modifications that we care about. */
+    if ((ret = __wt_config_gets(session, cfg + 1, "chunk_cache", &cval)) == WT_NOTFOUND)
+        return (0);
+
+    if (chunkcache->type == WT_CHUNKCACHE_UNCONFIGURED)
+        WT_RET_MSG(
+          session, EINVAL, "chunk cache reconfigure requested, but cache has not been configured");
+
+    WT_RET(__config_get_sorted_pinned_objects(session, cfg, &pinned_objects, &cnt));
+
+    /*
+     * Acquire the pinned array lock to avoid racing with threads reading the pinned array, and then
+     * update the array.
+     */
+    __wt_writelock(session, &chunkcache->pinned_objects.array_lock);
+    chunkcache->pinned_objects.array = pinned_objects;
+    chunkcache->pinned_objects.entries = cnt;
+    __wt_writeunlock(session, &chunkcache->pinned_objects.array_lock);
+
+    /* Release the memory allocated to the old array. */
+    __chunkcache_arr_free(session, &old_pinned_list);
```

Comment:
Why do we need a temporary for this? I'd have thought you could say

```
__wt_writelock(session, &chunkcache->pinned_objects.array_lock);
__chunkcache_arr_free(session, &chunkcache->pinned_object.array;
chunkcache->pinned_objects.array = pinned_objects;
```

Reply:
I did that to avoid work that could be done without taking a write lock. I don't mind doing either. Let me know which one you prefer.

Reply:
I'm with Sid on this one.  Minimize what's happening in the lock.  Free "should be" fast, but you never know.

Reply:
Fair enough then - I'll close this thread out.